framework/resources/Datasources/parquet_storage/filter/pushdown/varchar_decimal/copy_metadata.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+source conf/drillTestConfig.properties
+
+set -x
+
+test_dir=${DRILL_TEST_DATA_DIR}/Datasources/parquet_storage/filter/pushdown/varchar_decimal
+dfs_location=${DRILL_TESTDATA}/filter/pushdown/varchar_decimal
+
+copy_to_dfs() {
+    local metadata_file=$test_dir/$1
+    local destination=$dfs_location/$1
+
+    hadoop fs -rm $destination
+#   We need to copy the metadata files separately because otherwise Drill considers the tables as modified and auto refreshes metadata.
+#   Using "hadoop fs -put" doesn't help for some reason.
+    hadoop fs -touchz $destination
+    hadoop fs -appendToFile $metadata_file $destination
+}
+
+for file in $(find ${test_dir} -name .drill.parquet_metadata | grep -oE "[^/]*/[^/]*/[^/]*$")
+do
+    copy_to_dfs $file
```

Comment:
Trying to understand why we aren't using the framework's "cp" mode to copy over any test data. Can you try and see if that works?

Reply:
I'm using "cp" for parquet files, but if the metadata is being copied that way, Drill considers it obsolete (maybe due to the difference in modification time) and auto refresh it.

Reply:
Rahul had previously added tests which were copying over metadata cache files - both old and new. Can you look up to see how he achieved it?

Reply:
I looked on the data he added in dfs, and looks like all the metadata was refreshed by Drill. My purpose was to use metadata as is, without refreshing, because it is actually valid (it was generated at the same time as the tables). Getting the data from GitHub and then copying with "cp" is resulting in a mess with modification times and Drill then considers the tables as modified.
So it is required to copy metadata **after** the parquets.

Reply:
That was the first thing I tried, but it didn't work for me. I looked into the code and saw that cp tasks are being executed in parallel threads the same way as gen tasks:
`    final Stopwatch stopwatch = Stopwatch.createStarted();
    LOG.info("> Copying Data");
    copyExecutor.executeAll(copyTasks);
    copyExecutor.close();
    LOG.info(">> Copy duration: " + stopwatch + "\n");
    stopwatch.reset().start();
    LOG.info("> Generating Data");
    genExecutor.executeAll(genTasks);
    genExecutor.close();
    LOG.info("\n>> Generation duration: " + stopwatch + "\n");`
Regarding your second question, there was resent changes in statistics storing, so there are possible differences in behavior. For example, filter pushdown for decimals in old parquet files works only with old metadata. Another reason: metadata generated in prior versions of Drill should still be valid and queries that use it must not result in data loss or something. This case need to be covered.